drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -327,17 +327,37 @@ static void start_ble_rf(void)
 		LL_RCC_ReleaseBackupDomainReset();
 	}
 
-	/* Select LSE clock */
-	LL_RCC_LSE_Enable();
-	while (!LL_RCC_LSE_IsReady()) {
-	}
+	do {
+#ifdef CONFIG_CLOCK_STM32_LSE
+		const u32_t alarm = k_uptime_get_32() + K_MSEC(RCC_LSE_TIMEOUT_VALUE);
+		/* Select LSE clock */
+		LL_RCC_LSE_Enable();
+		while (!LL_RCC_LSE_IsReady()) {
+			if (alarm < k_uptime_get_32())
+				goto init_LSI;
+		}
+		/* Select wakeup source of BLE RF */
+		LL_RCC_SetRFWKPClockSource(LL_RCC_RFWKP_CLKSOURCE_LSE);
+		LL_RCC_SetRTCClockSource(LL_RCC_RTC_CLKSOURCE_LSE);
+
+		/* Switch OFF LSI */
+		LL_RCC_LSI1_Disable();
+		break;
+#endif
+init_LSI:
```

Comment:
#endif should be located after `init_LSI:`

Reply:
indeed. thanks